controllers/common/common.go

```diff
@@ -254,3 +313,68 @@ func TypeConverter(a, b interface{}) error {
 
 	return json.Unmarshal(js, b)
 }
+
+// Select objects that are deleted or created
+func GetAffectedObjs[T comparable](oldObjs []T, newObjs []T) []T {
+	table := make(map[T]int)
+
+	for _, oldObj := range oldObjs {
+		table[oldObj] = 1
+	}
+
+	for _, newObj := range newObjs {
+		table[newObj]++
+	}
+
+	result := []T{}
+
+	for key, val := range table {
+		if val == 1 {
+			result = append(result, key)
+		}
+	}
+
+	return result
+}
+
+type PlacementRefKinds string
+
+const (
+	Placement     PlacementRefKinds = "Placement"
+	PlacementRule PlacementRefKinds = "PlacementRule"
+)
```

Comment:
Do we handle any other case rather than placementRule or placement for PlacementRef?

Reply:
I guess I'm just saying I prefer the static check against a regular string.

Reply:
Before it was
 ```
		for i, pb := range pbList.Items {
			if pb.PlacementRef.APIGroup != appsv1.SchemeGroupVersion.Group ||
				pb.PlacementRef.Kind != "PlacementRule" || pb.PlacementRef.Name != object.GetName() {
				continue
			}

			result = append(result, common.GetPoliciesInPlacementBinding(ctx, c, &pbList.Items[i])...)
		}
```

Reply:
@yiraeChristineKim I think the issue is that it doesn't get you much since I can create a `PlacementRefKinds` with `PlacementRefKinds("some-random-string")`. That being said, I have seen this used in other libraries and I don't mind it, though it's more of a hint of accepted constants as opposed to any real validation.

Reply:
I thought golang enum more like typescript or Java enum. I didn't know that I could create any random string with that type. Thanks for letting me know